machine-learning-scikit-learn.ipynb

```diff
@@ -0,0 +1,333 @@
+   "source": [
+    "from dask_ml.model_selection import train_test_split\n",
+    "X_train, X_test, y_train, y_test = train_test_split(X, y)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Model creation"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's create an sklearn model as usual:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sklearn.linear_model import SGDClassifier\n",
+    "\n",
+    "est = SGDClassifier(loss='log', penalty='l1', tol=1e-3)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now, let's wrap it with `dask_ml.wrappers.Incremental`. This will allow this model to scale well to larger datasets that Dask holds."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from dask_ml.wrappers import Incremental\n",
+    "\n",
+    "inc = Incremental(est, scoring='accuracy')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Our model (SGDClassifier here) must implement `partial_fit` to be used with `Incremental`. A list of models that implement this API is available on a scikit-learn documentation page under \"Incremental learning\": http://scikit-learn.org/stable/modules/scaling_strategies.html#incremental-learning\n",
+    "\n",
+    "`Incremental` does data management: it calls `est.partial_fit` on each chunk of the passed data. Dask moves the model to different chunks of the data (or vice versa) to let this happen. Note: Dask-ML `Incremental` gives data to Scikit-learn and does not change the underlying optimization algorithm that Scikit-learn uses. There are different optimization algorithms in Dask-ML that do this, especially for clustering and matrix decomposition.\n",
+    "\n",
+    "It is important to specify the scoring parameter in `Incremental`; otherwise, scikit-learn scorers are fed Dask arrays (which they're not optimized for).\n",
+    "\n",
+    "\n",
+    "[1]:http://scikit-learn.org/stable/modules/scaling_strategies.html\n",
+    "[inc]:http://scikit-learn.org/stable/modules/scaling_strategies.html#incremental-learning"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Model training\n",
+    "For an `Incremental` model, `partial_fit` and `fit` aliased to the same item and do one complete pass over the dataset. Let's call `partial_fit` a couple times, and score it after it's called each time (which will allow for a visualization):"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%time\n",
+    "# takes about 1min\n",
+    "\n",
+    "from distributed.metrics import time\n",
+    "data = []\n",
+    "start = time()\n",
+    "for iteration in range(10):\n",
```

Comment:
It would also be good to note (somewhere) that these multiple epochs + distributed provide an advantage over writing your own for loop to load in chunks of data off disk: With distributed you only pay the IO cost once.